ckanext/syndicate/tasks.py

```diff
@@ -161,7 +213,17 @@ def _update_package(package):
 
         updated_package['extras'] = filter_extras(package['extras'])
         updated_package['resources'] = filter_resources(package['resources'])
-        updated_package['owner_org'] = get_syndicated_organization()
+
+        org = updated_package.pop('organization')
+
+        try:
+            if not is_organization_preserved():
```

Comment:
I wasn't explicit about this before, but could you change this as you have done with lines 149-152